### PR TITLE
Validate summary results

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- https://quartech.atlassian.net/browse/BHBC-1318

## This PR contains the following changes

- uploading summary results
- validating against validation rules
- if validation rules pass, scrape and insert the content in the DB

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/paths/project/{projectId}/survey/{surveyId}/summary/submission/upload.ts [hunk 1]
```diff
@@ -249,6 +262,272 @@ export const updateSurveySummarySubmissionWithKey = async (
...
+function persistSummaryParseErrors(): RequestHandler {
+  return async (req, res, next) => {
+    const parseError = req['parseError'];
+
+    if (!parseError) {
+      // no errors to persist, skip to next step
+      return next();
+    }
+
+    defaultLog.debug({ label: 'persistSummaryParseErrors', message: 'parseError', parseError });
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/survey/{surveyId}/summary/submission/upload.ts]: this should be before the if statement right

**Review by sdevalapurkar — COMMENTED** [review]
overall looking good to me

File: api/src/paths/project/{projectId}/survey/{surveyId}/summary/submission/upload.ts (hunk 1 above)
Comment (anissa-agahchen) [reply api/src/paths/project/{projectId}/survey/{surveyId}/summary/submission/upload.ts]: nice catch!  Thanks

File: api/src/queries/survey/survey-summary-queries.ts [hunk 2]
```diff
@@ -70,28 +70,36 @@ export const getLatestSurveySummarySubmissionSQL = (surveyId: number): SQLStatem
...
+    sssm.submission_message_type_id = ssmt.submission_message_type_id
+  LEFT OUTER JOIN
+    summary_submission_message_class as ssmc
+  ON
+    ssmt.summary_submission_message_class_id = ssmc.summary_submission_message_class_id
+  WHERE
+    sss.survey_id = ${surveyId}
+  ORDER BY
+    sss.event_timestamp DESC
+  LIMIT 1;
```
Comment (cgarrettjones) [inline api/src/queries/survey/survey-summary-queries.ts:101]: always curious about LIMIT queries. if the data should have only one record then not letting it blow up hides a bug somewhere else?

**Comment by cgarrettjones** [comment]
overall looks good!

File: api/src/queries/survey/survey-summary-queries.ts (hunk 2 above)
Comment (anissa-agahchen) [reply api/src/queries/survey/survey-summary-queries.ts:101]: we expect it to have multiple records, and want the latest based on timestamp.

Comment (cgarrettjones) [reply api/src/queries/survey/survey-summary-queries.ts:101]: cool. thanks for clarifying.

**Review by NickPhura — APPROVED** [review]

**Pull request merged**